Mathlib/Algebra/Module/Submodule/Map.lean

```diff
@@ -208,6 +208,27 @@ theorem le_comap_pow_of_le_comap (p : Submodule R M) {f : M →ₗ[R] M}
   | zero => simp [Module.End.one_eq_id]
   | succ k ih => simp [Module.End.iterate_succ, comap_comp, h.trans (comap_mono ih)]
 
+section comapRestrict
+
+variable [Module R M₂] (q : Submodule R M₂) (f : M →ₗ[R] M₂)
+
+/-- For a linear map `f`, the map from `Submodule.comap q f` to `q` for a submodule `q`
+obtained from restricting `f` using `LinearMap.restrict`. -/
+def comapRestrict : ↥(q.comap f) →ₗ[R] ↥q :=
```

Comment:
`comapRestrict`  is a particular case of `restrict`: what about just using its definition (maybe as an `abbrev`) without introducing a new def?

Reply:
Done. Now it's `abbrev` instead of `def` so the simp lemma is not needed anymore.

BTW, I split this to a small PR #24451. Hopefully it will speed up the reviewing process.